src/solvers/dgsem_tree/dg_2d.jl

```diff
@@ -490,6 +512,144 @@ end
 end
 
 
+function calc_volume_integral!(du, u,
+                               mesh::TreeMesh{2},
+                               nonconservative_terms, equations,
+                               volume_integral::VolumeIntegralStaggeredGrid,
+                               dg::DGSEM, cache)
+  @threaded for element in eachelement(dg, cache)
+    staggered_grid_kernel!(du, u, element, mesh,
+                           nonconservative_terms, equations,
+                           volume_integral, dg, cache)
+  end
+end
+
+@inline function staggered_grid_kernel!(du, u,
+                                        element, mesh::TreeMesh{2},
+                                        nonconservative_terms::Val{false}, equations,
+                                        volume_integral, dg::DGSEM, cache, alpha=true)
+  # true * [some floating point value] == [exactly the same floating point value]
+  # This can (hopefully) be optimized away due to constant propagation.
+  @unpack inverse_weights = dg.basis
+  @unpack volume_flux_dg, volume_flux_fv, indicator = volume_integral
+
+  # high-order dg fluxes
+  @unpack fstaggered1_threaded, fstaggered2_threaded = cache
+
+  fstaggered1 = fstaggered1_threaded[Threads.threadid()]
+  fstaggered2 = fstaggered2_threaded[Threads.threadid()]
+  calcflux_staggered!(fstaggered1, fstaggered2, u, mesh,
+                      nonconservative_terms, equations, volume_flux_dg, dg, element, cache)
+
+  # low-order fv fluxes
+  @unpack fstar1_L_threaded, fstar1_R_threaded, fstar2_L_threaded, fstar2_R_threaded = cache
+
+  fstar1_L = fstar1_L_threaded[Threads.threadid()]
+  fstar2_L = fstar2_L_threaded[Threads.threadid()]
+  fstar1_R = fstar1_R_threaded[Threads.threadid()]
+  fstar2_R = fstar2_R_threaded[Threads.threadid()]
+  calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u, mesh,
+               nonconservative_terms, equations, volume_flux_fv, dg, element, cache)
+
+
+  # Calculate blending factor alpha_blending
+  alpha_blending = volume_integral.indicator
```

Comment:
This needs to be computed somehow, doesn't it?

Reply:
Yes. Right now, we're getting the general idea of different indicators, which I will implement later on.